Leanplum-SDK/Classes/Managers/Networking/LPRequestSender.m

```diff
@@ -148,81 +161,15 @@ - (void)sendEventually:(LPRequest *)request sync:(BOOL)sync
     [self.countAggregator incrementCount:@"send_eventually_lp"];
 }
 
-- (void)sendIfConnected:(LPRequest *)request
-{
-    LP_TRY
-    [self sendIfConnected:request sync:NO];
-    LP_END_TRY
-    
-    [self.countAggregator incrementCount:@"send_if_connected_lp"];
-}
-
-- (void)sendIfConnected:(LPRequest *)request sync:(BOOL)sync
-{
-    if ([[Leanplum_Reachability reachabilityForInternetConnection] isReachable]) {
-        if (sync) {
-            [self sendNowSync:request];
-        } else {
-            [self sendNow:request];
-        }
-    } else {
-        [self sendEventually:request sync:sync];
-        if (request.errorBlock) {
-            request.errorBlock([NSError errorWithDomain:@"Leanplum" code:1
-                                               userInfo:@{NSLocalizedDescriptionKey: @"Device is offline"}]);
-        }
-    }
-    
-    [self.countAggregator incrementCount:@"send_if_connected_sync_lp"];
-}
-
 - (void)sendNow:(LPRequest *)request
 {
-    [self sendNow:request sync:NO];
-}
-
-- (void)sendNowSync:(LPRequest *)request
-{
-    [self sendNow:request sync:YES];
-}
-
-// Wait 1 second for potential other API calls, and then sends the call synchronously
-// if no other call has been sent within 1 minute.
-- (void)sendIfDelayed:(LPRequest *)request
-{
-    [self sendEventually:request sync:NO];
-    [self performSelector:@selector(sendIfDelayedHelper:)
-               withObject:request
-               afterDelay:LP_REQUEST_RESUME_DELAY];
-
-}
-
-// Sends the call synchronously if no other call has been sent within 1 minute.
-- (void)sendIfDelayedHelper:(LPRequest *)request
-{
-    LP_TRY
-    if ([LPConstantsState sharedState].isDevelopmentModeEnabled) {
-        [self send:request];
-    } else {
-        NSTimeInterval currentTime = [NSDate timeIntervalSinceReferenceDate];
-        if (!self.lastSentTime || currentTime - self.lastSentTime > LP_REQUEST_PRODUCTION_DELAY) {
-            [self sendIfConnected:request];
-        }
-    }
-    LP_END_TRY
-}
-
-- (void)sendNow:(LPRequest *)request withData:(NSData *)data forKey:(NSString *)key
-{
-    [self sendNow:request withDatas:@{key: data}];
-    
-    [self.countAggregator incrementCount:@"send_now_with_data_lp"];
+    [self sendNow:request sync:request.sendSynchronously];
```

Comment:
Please get rid off `sync` flag from all methods and always execute on `LPOperationQueue`